transaction/src/transaction_view/action_view.rs

```diff
@@ -0,0 +1,17 @@
+pub mod output_view;
+pub mod spend_view;
+pub mod swap_claim_view;
+pub mod swap_view;
+
+pub use output_view::OutputView;
+pub use spend_view::SpendView;
+pub use swap_claim_view::SwapClaimView;
+pub use swap_view::SwapView;
+
+pub enum ActionView {
+    /// View of a Swap action
+    Swap(SwapView),
+    SwapClaim(SwapClaimView),
+    Output(OutputView),
+    Spend(SpendView),
+}
```

Comment:
This only covers actions that have decryptable content. How does a consumer of a `TransactionView` understand a transaction, like an ICS20 withdrawal, that includes a mix of shielded and transparent actions?

Reply:
Latest design will associate each `Action` with a corresponding `ActionView` so that a consumer can treat shielded & transparent actions as the same type of thing.